src/Pages/Detail/Review.js

```diff
@@ -1,188 +1,156 @@
-import "./Review.scss";
-
+import React, { Component } from 'react';
+import CommentList from './CommentList'
+import './Review.scss';
+ 
+ 
 class Review extends Component {
-  constructor() {
-    super();
-    this.state = {
-      user: "김영*",
-      text: "",
-      replies: [
-        {
-          name: "",
-          comment: "",
-        },
-      ],
-      buttonName: "댓글을 작성해주세요",
-      show: true,
-      totalScore: 0,
-      score: 0,
-    };
-  }
-  // num=0
-
-  handleText = (e) => {
-    this.setState({
-      text: e.target.value,
-    });
-  };
-
-  addText = (e) => {
-    let arr = this.state.replies;
-    arr.push({
-      name: this.state.user,
-      comment: this.state.text,
-    });
-    this.setState({
-      replies: arr,
-      text: "",
-    });
-  };
+    
+    constructor(){
+        super();
+        this.state = {
+            user:"김영*",
+            text: "",
+            replies:[{
+                name:"",
+                comment:""
+            }],
+            show: true,
+            buttonName: "댓글을 작성해주세요",
+            yesClicked: true,
+            noClicked: false,
+        };
+    }   
 
-  pressEnter = (e) => {
-    if (e.key === "Enter" && this.state.text) {
-      this.addText();
-      e.target.value = "";
+    
+    handleText = (e) => {
+     this.setState({
+        text: e.target.value
+     });    
     }
-  };
 
-  handleRemove = (num) => {
-    alert("이 댓글을 정말로 삭제하시겠습니까?");
-    const { replies } = this.state;
-    const nextReply = replies.filter((text) => {
-      return text.num !== num;
-    });
-    this.setState({
-      replies: nextReply,
-    });
-  };
+    addText = (e) => {
+       let arr = this.state.replies;
+       arr.push({
+           name: this.state.user,
+           comment: this.state.text
+       });
+       this.setState({
+           replies: arr,
+           text: "",
+       })
+    }
+    
+    pressEnter=(e) =>{
+        if(e.key ==="Enter" && this.state.text) {
+            this.addText();
+            e.target.value="";
+        }
+    }
 
-  showAllComment = () => {
-    if (this.state.show) {
-      this.setState({
-        show: false,
-        buttonName: "댓글접기",
-      });
-    } else if (!this.state.show) {
-      this.setState({
-        show: true,
-        buttonName: "댓글을 작성해주세요",
-      });
+    handleRemove=(num)=>{
+        alert('이 댓글을 정말로 삭제하시겠습니까?')
+        const{ replies } =this.state;
+        const nextReply = replies.filter((text,index)=>{
+            return index!== num;
+        });
+        this.setState({
+            replies: nextReply,
+        })
+    }
+   
+    showAllComment =() =>{
+        if(this.state.show){
+            this.setState({
+                show: false,
+                buttonName: "댓글접기"
+            })
+            }else if (!this.state.show){
+                this.setState({
+                    show: true,
+                    buttonName: "댓글을 작성해주세요"
+                })
+            }
+        }
```

Comment:
```suggestion
    showAllComment = () =>{
       const { show } = this.state
       this.setState({
         show: !show,
         buttonName: show ? "댓글접기" : "댓글을 작성해주세요"
       })
```
setState는 대부분의 경우 1번만 적어줘도 됩니다